homeassistant/components/nmap_tracker/device_tracker.py

```diff
@@ -31,6 +32,7 @@
         vol.Required(CONF_HOME_INTERVAL, default=0): cv.positive_int,
         vol.Optional(CONF_EXCLUDE, default=[]): vol.All(cv.ensure_list, [cv.string]),
         vol.Optional(CONF_OPTIONS, default=DEFAULT_OPTIONS): cv.string,
+        vol.Optional(CONF_SUDO, default=False): cv.boolean,
```

Comment:
Why not detect if this is needed, instead of a config option?

Furthermore, we don't allow changes to the YAML anymore.

See: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

This integration needs some refactoring before this can be added. Feel free to start on that! 👍

Reply:
How would you propose detecting it? For security reasons, I'm not sure if we should default to sudo on Mac; it might not work (it'd need NOPASSWD in sudoers, I assume). On Linux, adding `--privileged` and `setcap` are optional; there are cases where someone might not feel comfortable doing it. 

I didn't realize that yaml changes were no longer allowed. I think the migration to the newer format might be a bit past what I have time to commit to, but if someone else does, perhaps we can reference this PR for a useful bit of functionality to add. I suppose for now, I'll just keep this as a local patch (to be fair, Mac OS represents a rather small percentage of HA users, and I assume you probably want to see people moving more towards docker and the more standard installation types (sorry, I didn't quite keep up with the whole HA->Core / Hass etc rebranding, as it didn't affect me personally :-) )

Reply:
I guess one could check if the capabilities are available, and decide what to do.

This is not about directing people to the different installation methods, Home Assistant does support running the Core on macOS (see: <https://github.com/home-assistant/architecture/blob/master/adr/0016-home-assistant-core.md>)

Reply:
I'm not comfortable, personally, with defaulting to sudo if extra capabilities aren't available, whether because it's not supported (Mac) or not done explicitly (Linux). From a security POV, I would not want to do this. 

And just to be clear, I meant that this bug affects a small minority of users, not that it's unofficial or unsupported. It was more my excuse for not wanting to tackle a bigger rewrite of something I don't quite understand than anything else.

Thanks for the feedback. I'm afraid there's probably nothing I can do at this point to push it forward. I'm going to close this PR, but maybe it will help someone else in the future.